src/robot/parsing/model.py

```diff
@@ -595,20 +595,33 @@ def __iter__(self):
 
 
 class ForLoop(_WithSteps):
+    """
+    Properties:
+
+    - flavor: The value of the 'IN' item, with spaces removed.
+              Typically 'IN', 'INRANGE', 'INZIP', 'INEUMERATE', etc.(?)
+    - range:  True if flavor is 'INRANGE'
+    - vars:   Variables set per-iteration by this loop
+    - items:  Items to loop over (or otherwise parameters, like with INRANGE)
+    - comment: a comment, or None.
+    - steps:  A list of steps in the loop.
+    """
 
     def __init__(self, declaration, comment=None):
-        self.range, index = self._get_range_and_index(declaration)
+        self.flavor, index = self._get_flavor_and_index(declaration)
+        # self.range is preserved for backwards-compatibility
+        self.range = self.flavor == 'INRANGE'
```

Comment:
Done, though I'm not totally happy with how it turned out: I had to add another property (tentatively called `raw_flavor`) to preserve any spaces in the original name.

...actually, I'm going to drop the spaces-removed version of `flavor` and only remove those spaces in the dispatch-factory.